vm-virtio/src/console.rs

```diff
@@ -380,15 +379,14 @@ impl Console {
             kill_evt: None,
             avail_features,
             acked_features: 0u64,
-            config_space: Vec::new(),
+            config_space: console_config.lock().unwrap().as_slice().to_vec(),
```

Comment:
So here you're introducing a bug IMO. By doing so, you're copying the `VirtioConsoleConfig` based on its value at this specific time, but later when the `read_config` is being triggered, it will read the same value since your copy cannot change.
Problem is, the `virtio-console` device can change its config at runtime, depending on the size of the terminal. This means the `VirtioConsoleConfig` is being modified and its values are being changed.
You need to handle this properly, otherwise we won't be able to resize the terminal with `virtio-console` anymore.
I'm surprised the CI didn't catch that as I thought we had a test covering this.

Reply:
@sboeuf Yes, there will be a bug. To avoid such issue, it seems that config should not be removed, it could be used to transfer the actual config data to config_space in VirtioBackend while read_config is called. Please take a look at the latest code about this.

Reply:
@bjzhjing Looking at the way you handled this, I realized there's no simple way to solve this problem without involving a lot of refactoring for the `virtio-console` device.
But if we look at the problem differently, I would say that virtio devices (not all of them but some) might need update their configuration at runtime, coming from the user. A good example of that might be `virtio-blk`, which can be updated with a different image at runtime.
Based on this observation, I think we should change the `VirtioBackend` structure so that `config_space` field would be an `Arc<Mutex<Vec<u8>>>`. This is the generic way to handle the fact that a config can be updated and this should be reflected through `VirtioBackend` struct.
/cc @sameo @rbradford

Reply:
@sboeuf please take a look, Arc Mutex is added to protect config_space in VirtioBackend.